src/censure.jl

```diff
@@ -0,0 +1,236 @@
+abstract BiFilter
+
+type BoxFilter <: BiFilter
+
+	scale :: Integer
+	in_length :: Integer
```

Comment:
Note that this only applies to types in fields, not arguments in functions (there is no penalty for declaring an argument as `dim::Integer`). The reason is that a type-declaration is *prospective* information for use in inference; in contrast, method specialization uses the actual types of arguments when called.